cocos/ui/UIRichText.cpp

```diff
@@ -1800,12 +1825,34 @@ void RichText::formarRenderers()
     }
     updateContentSizeWithTextureSize(_contentSize);
 }
-    
+
+void RichText::doHorizontalAlignment(const Vector<cocos2d::Node*> &row, float rowWidth) {
+    const auto alignment = static_cast<HorizontalAlignment>(_defaults.at(KEY_HORIZONTAL_ALIGNMENT).asInt());
+    if ( alignment != HorizontalAlignment::LEFT ) {
+        const auto leftOver = getContentSize().width - rowWidth;
+        const float leftPadding = [alignment, leftOver] {
+            switch ( alignment ) {
+                case HorizontalAlignment::CENTER:
+                    return leftOver / 2.f;
+                case HorizontalAlignment::RIGHT:
+                    return leftOver;
+                default:
+                    CCASSERT(false, "invalid horizontal alignment!");
+                    return 0.f;
+            }
+        }();
```

Comment:
You're right, it's not strictly necessary.  The immediately-invoked lambda pattern is useful for complex initialization of variables, which, after initialization, should then be `const`. http://www.bfilipek.com/2016/11/iife-for-complex-initialization.html

Without a lambda, it would be difficult to make the `leftPadding` variable `const`.  Maybe that's not a big deal; if it's too weird, I can change it.